viewer/src/main/webapp/viewer-html/components/OpenLayersSnappingController.js

```diff
@@ -0,0 +1,259 @@
+ * @author <a href="mailto:markprins@b3partners.nl">Mark Prins</a>
+ *
+ * @class
+ */
+Ext.define("viewer.components.OpenLayersSnappingController", {
+    extend: "viewer.components.Component",
+    /**
+     * editable/drawable OpenLayers Vector layer.
+     * @private
+     * @TODO refactor to array as each edit & draw control have their own layer
+     */
+    frameworkLayer: null,
+    /** snapping control.*/
+    frameworkControl: null,
+    config: {
+        style: {
+            strokeColor: '#FF00FF',
+            strokeOpacity: 0.5,
+            strokeWidth: 1,
+            pointRadius: 1,
+            fillOpacity: 0.25,
+            fillColor: '#FF00FF'
+        },
+        viewerController: null
+    },
+    /**
+     * OpenLayers Vector layers to snap to.
+     * @private
+     */
+    snapLayers: [],
+    /**
+     * name prefix of the built-in snapLayers
+     */
+    snapLayers_prefix: "snapping_",
+    /**
+     * OpenLayers Map.
+     * @private
+     */
+    frameworkMap: null,
+    /**
+     * @constructor
+     * @param {Object} config
+     * @returns {viewer.viewercontroller.openlayers.OpenLayersSnappingLayer|OpenLayersSnappingControllerAnonym$0.constructor}
+     */
+    constructor: function (config) {
+        viewer.components.OpenLayersSnappingController.superclass.constructor.call(this, config);
+        this.frameworkMap = this.config.viewerController.mapComponent.getMap().getFrameworkMap();
+        this.frameworkControl = new OpenLayers.Control.Snapping();
+
+        this.config.viewerController.mapComponent.getMap().addListener(
+                viewer.viewercontroller.controller.Event.ON_LAYER_REMOVED,
+                this.layerRemoved, this);
+        this.config.viewerController.mapComponent.getMap().addListener(
+                viewer.viewercontroller.controller.Event.ON_LAYER_ADDED,
+                this.layerAdded, this);
+        this.config.viewerController.mapComponent.getMap().addListener(
+                viewer.viewercontroller.controller.Event.ON_FINISHED_CHANGE_EXTENT,
+                this.changedExtent, this);
+
+        return this;
+    },
+    /**
+     * @param {type} map
+     * @param {type} options
+     */
+    layerRemoved: function (map, options) {
+        if (options.layer.getType() !== "VECTOR") {
+            return;
+        }
+        // assume we now have a drawing or editing layer
+        //this.getlayerName(options.layer.appLayerId);
+        // TODO
+        // look up the control that belongs to this appLayer and destroy
+    },
+    /**
+     * attach the snapping control to the Openlayers layers of the added editing or drawing layer.
+     * @param {type} map
+     * @param {type} options
+     */
+    layerAdded: function (map, options) {
+        if (options.layer.getType() !== "VECTOR") {
+            return;
+        }
+        // filter for edit and drawing control layers
+        if ((Ext.String.startsWith(options.layer.name, "drawing", true)) ||
```

Comment:
Seems fragile to depend on the layername, since it can change. Don't have any better ideas though..